src/connectionimpl.cpp

```diff
@@ -123,7 +123,9 @@ uint64_t ConnectionImpl::parse(const Buffer &buffer)
         try
         {
             // try to recognize the frame
-            ReceivedFrame receivedFrame(ReducedBuffer(buffer, processed), _maxFrame);
```

Comment:
I'm sorry, but that example looks irrelevant to me (there's nothing about saving a reference).
C++ expressions can't consist of several statements (not willing to quote Standard for you, sorry again)

Actually, you could compile that code snippet with an existing compiler and see the output.
http://pastebin.com/UcLrCkvL